src/rust/rust-proto/src/protocol/service_client.rs

```diff
@@ -45,3 +51,32 @@ impl From<client_executor::Error<ConnectError>> for ConnectError {
         }
     }
 }
+
+#[async_trait::async_trait]
+pub trait ConfigConnectable
+where
+    Self: Connectable,
+{
+    type Config: GrpcClientConfig + Send + Sync;
+}
+
+#[async_trait::async_trait]
+pub trait ConnectWithConfig
```

Comment:
Same question--is there ever a case where a `Connectable` _doesn't_ have a config? In that case, it seems like all this stuff should just go on `Connectable`.

Reply:
I believe the answer is "potentially"; I'd previously thought about the following:
```

fn get_plugin_upstream_address(plugin_id: uuid::Uuid) -> String {
    let upstream_addr_env_var = format!("NOMAD_UPSTREAM_ADDR_plugin-{plugin_id}");
    let upstream_addr = std::env::var(&upstream_addr_env_var).expect(&upstream_addr_env_var);
    let address = format!("http://{upstream_addr}");
    address
}

/// Create a client from environment
pub async fn get_generator_client(
    plugin_id: uuid::Uuid,
) -> Result<GeneratorServiceClient, ConnectError> {
    let address = get_plugin_upstream_address(plugin_id);
    let client_config = GeneratorClientConfig {
        generator_client_address: address.parse().expect("generator_client_address"),
    };
    GeneratorServiceClient::connect_with_config(client_config).await
    ```
    
  where we manually construct the Config instead of doing anything clap/parse-y with it